packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -675,98 +703,99 @@ - (NSDictionary *)encodingMap
-  sessionConfiguration.requestCachePolicy = NSURLRequestReloadIgnoringLocalCacheData;
-  sessionConfiguration.URLCache = nil;
-  
-  __weak typeof(self) weakSelf = self;
-  EXDownloadDelegateOnWriteCallback onWrite = ^(NSURLSessionDownloadTask *task, int64_t bytesWritten, int64_t totalBytesWritten, int64_t totalBytesExpectedToWrite) {
-    __strong EXFileSystem *strongSelf = weakSelf;
-    if (strongSelf && bytesWritten > 0) {
-      [strongSelf sendEventWithName:EXDownloadProgressEventName
-                               body:@{@"uuid":uuid,
-                                      @"data":@{
-                                          @"totalBytesWritten": @(totalBytesWritten),
-                                          @"totalBytesExpectedToWrite": @(totalBytesExpectedToWrite),
-                                          },
-                                      }];
-    }
-  };
-  
-  EXDownloadDelegateOnDownloadCallback onDownload = ^(NSURLSessionDownloadTask *task, NSURL *location) {
-    NSURL *scopedLocation = [NSURL fileURLWithPath:scopedPath];
-    NSData *locationData = [NSData dataWithContentsOfURL:location];
-    [locationData writeToFile:scopedPath atomically:YES];
-    NSData *data = [NSData dataWithContentsOfURL:scopedLocation];
-    if (!data) {
-      reject(@"E_UNABLE_TO_SAVE",
-             @"Unable to save file to local URI",
-             nil);
-      return;
-    }
-    NSMutableDictionary *result = [NSMutableDictionary dictionary];
-    result[@"uri"] = scopedLocation.absoluteString;
-    result[@"complete"] = @(YES);
-    if (options[@"md5"]) {
-      result[@"md5"] = [data md5String];
+  for (id key in [headers allKeys]) {
+    if (![key isKindOfClass:[NSString class]] || ![headers[key] isKindOfClass:[NSString class]]) {
+      return false;
     }
-    NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)task.response;
-    result[@"status"] = @([httpResponse statusCode]);
-    result[@"headers"] = [httpResponse allHeaderFields];
-    
-    __strong EXFileSystem *strongSelf = weakSelf;
-    if (strongSelf) {
-      [strongSelf.downloadObjects removeObjectForKey:uuid];
-    }
-    
-    resolve(result);
-  };
-  
-  EXDownloadDelegateOnErrorCallback onError = ^(NSError *error) {
-    //"cancelled" description when paused.  Don't throw.
-    if ([error.localizedDescription isEqualToString:@"cancelled"]) {
-      __strong EXFileSystem *strongSelf = weakSelf;
-      if (strongSelf) {
-        [strongSelf.downloadObjects removeObjectForKey:uuid];
-      }
-      resolve([NSNull null]);
-    } else {
-      reject(@"E_UNABLE_TO_DOWNLOAD",
-             [NSString stringWithFormat:@"Unable to download from: %@", url.absoluteString],
-             error);
-    }
-  };
+  }
   
+  return true;
+}
+
+- (NSURLSession *)_createSession:(id<NSURLSessionDelegate>)delegate options:(NSDictionary *)options
+{
+  EXFileSystemSessionType sessionType = [options[@"sessionType"] intValue] ?: EXFileSystemBackgroundSession;
```

Comment:
I don't think that making it foreground as the default is a good idea. It will create a difference between platforms, which may confuse a lot of people.